peps/pep-0705.rst

```diff
@@ -118,175 +116,84 @@ However, this no longer works once you start nesting dictionaries::
 
-    def get_name(entertainment: Entertainment) -> str:
-        if "movie" in entertainment:
-            return entertainment["movie"]["name"]
-        elif "book" in entertainment:
-            return entertainment["book"]["name"]
-        else:
-            raise ValueError("Unexpected entertainment type")
+This PEP only proposes making ``ReadOnly`` valid in a ``TypedDict``. A possible future extension would be to support it in additional contexts, such as in protocols.
 
-Note this is a subset of the functionality of the `unmerged proposal of PEP-728 <https://github.com/python/peps/pull/3441>`_.
 
 Specification
 =============
 
-``TypedDict`` will gain two new flags: ``other_keys`` and ``readonly``. A new ``typing.ReadOnly`` type qualifier is added.
-
-``other_keys`` flag
--------------------
-
-The optional ``other_keys`` flag to ``TypedDict`` can have the value ``typing.Never``, indicating that instances may only contain keys explicitly listed in the type::
-
-    class Album(TypedDict, other_keys=Never):
-        name: str
-        year: int
-
-    class AlbumExtra(Album, TypedDict):
-        band: str  # Runtime error
-
-Type-checkers may rely on this restriction::
-
-    def album_keys(album: Album) -> Collection[Literal['name', 'year']]:
-        # Type checkers may permit this, but should error if Album did not specify `other_keys=Never`
-        return album.keys()
-
-Type-checkers should prevent operations that would violate this restriction::
-
-    class AlbumExtra(TypedDict, other_keys=Never):
-        name: str
-        year: int
-        band: str
-
-    album: AlbumExtra = { "name": "Flood", year: 1990, band: "They Might Be Giants" }
-    album_keys(album)  # Type check error: extra key 'band'
-
-This PEP does not propose supporting any other values than ``other_keys=Never``. Future or concurrent PEPs may extend this flag to permit other types.
-
-``readonly`` flag
------------------
-
-The optional boolean ``readonly`` flag to ``TypedDict``, when ``True``, indicates that no mutator operations (``__setitem__``, ``__delitem__``, ``update``, etc.) will be permitted::
-
-    class NamedDict(TypedDict, readonly=True):
-        name: str
-
-    def get_name(d: NamedDict) -> str:
-        return d["name"]
-
-    def set_name(d: NamedDict, name: str) -> None:
-        d["name"] = name  # Type check error: cannot modify a read-only entry
-
-The ``readonly`` flag defaults to ``False``.
+A new ``typing.ReadOnly`` type qualifier is added.
 
 ``typing.ReadOnly`` type qualifier
 ----------------------------------
 
-The ``typing.ReadOnly`` type qualifier is used to indicate that a variable declared in a ``TypedDict`` definition may not be mutated by any operation performed on instances of the ``TypedDict``::
+The ``typing.ReadOnly`` type qualifier is used to indicate that an item declared in a ``TypedDict`` definition may not be mutated (added, modified, or removed) by any operation performed on variables of that type::
```

Comment:
This doesn't seem right: the instance is *not* of the type, as it's a structural type. It's a dict, and the items *can* be mutated on the dict instance. Perhaps there is another term that could fit here?

Reply:
How about "values"?

Reply:
That sounds too close to the values _in_ the dictionary, but also I think it has the same confusion.

It's really the type of the variable we care about here, not the thing it holds.

Perhaps there's a way to rewrite the whole sentence to avoid the need to distinguish this at all?

Reply:
```suggestion
The ``typing.ReadOnly`` type qualifier is used to indicate that an item declared in a ``TypedDict`` definition may not be mutated (added, modified, or removed)::
```
I think this is enough to get the point across.

Reply:
What do you think about the expression "through this type"? "may not be mutated...through this type"? I don't want to give the impression the item can't be mutated *at all*, if there's a clean way to express it.